financial-connections/src/main/java/com/stripe/android/financialconnections/features/generic/GenericScreen.kt

```diff
@@ -0,0 +1,345 @@
+import com.stripe.android.financialconnections.ui.theme.FinancialConnectionsTheme.typography
+import com.stripe.android.financialconnections.ui.theme.Layout
+import com.stripe.android.uicore.image.StripeImage
+import androidx.compose.ui.Alignment as ComposeAlignment
+
+@Preview
+@Composable
+internal fun GenericScreenPreview(
+    @PreviewParameter(GenericScreenPreviewParameterProvider::class) state: GenericScreenState
+) {
+    FinancialConnectionsPreview {
+        Surface(color = colors.backgroundSurface) {
+            GenericScreen(
+                state = state,
+                onClickableTextClick = {},
+                onSecondaryButtonClick = {},
+                onPrimaryButtonClick = {},
+            )
+        }
+    }
+}
+
+@Composable
+internal fun GenericScreen(
+    state: GenericScreenState,
+    onPrimaryButtonClick: () -> Unit,
+    onSecondaryButtonClick: () -> Unit,
+    onClickableTextClick: (String) -> Unit,
+) {
+    val density = LocalDensity.current
+
+    var containerHeight by remember { mutableStateOf(0.dp) }
+    var footerHeight by remember { mutableStateOf(0.dp) }
+    var contentHeight by remember { mutableStateOf(0.dp) }
+
+    val spacing by remember {
+        derivedStateOf {
+            ((containerHeight - footerHeight - contentHeight) / 2).takeIf {
+                state.screen.options?.verticalAlignment == VerticalAlignment.Centered && it > 0.dp
+            }
+        }
+    }
+
+    Layout(
+        bodyPadding = PaddingValues(vertical = 16.dp),
+        inModal = state.inModal,
+        footer = state.screen.footer?.let {
+            {
+                GenericFooter(
+                    payload = it,
+                    modifier = Modifier.onGloballyPositioned {
+                        with(density) {
+                            footerHeight = it.size.height.toDp()
+                        }
+                    },
+                    onPrimaryButtonClick = onPrimaryButtonClick,
+                    onSecondaryButtonClick = onSecondaryButtonClick,
+                    onClickableTextClick = onClickableTextClick,
+                )
+            }
+        },
+        modifier = Modifier.onGloballyPositioned {
+            with(density) {
+                containerHeight = it.size.height.toDp()
+            }
+        }
+    ) {
+        spacing?.let {
+            Spacer(modifier = Modifier.height(it))
+        }
+
+        Column(
+            modifier = Modifier.onGloballyPositioned {
+                with(density) {
+                    contentHeight = it.size.height.toDp()
+                }
+            }
+        ) {
+            state.screen.header?.let {
+                GenericHeader(
+                    payload = it,
+                    onClickableTextClick = onClickableTextClick,
+                    modifier = Modifier.padding(horizontal = 24.dp),
+                )
+            }
+
+            Spacer(modifier = Modifier.height(20.dp))
```

Comment:
Can we use `Arrangment.spacedBy()` for this? This way, it’ll only be rendered if both a header and a body are present.